### ISSUE TITLE
Blade component casting your enums automagically to string always

### ISSUE THREAD

**Issue by torian257x** [BenSampo/laravel-enum#229]
```

class Section extends Component
{

    public SectionColorType         $section_color;
```

in the view, when I access $section_color its a string. 

this is a major let down as I planned to use in my blade component

mycomponent.blade.php
```
<div class="$section_color->getCssClasses()">
```

who is doing this cast to string magic an why?

happens as well when "manually" passing it

e.g.

```

    public function render()
    {
        return view('components.section', ['x' => new SectionColorType('section_orange')]);
    }
```

`dd($x); //returns "section_orange"`

**Comment by BenSampo** [comment]
Hi @torian257x.

`Enum` classes implement the [`__toString()` magic method](https://www.php.net/manual/en/language.oop5.magic.php#object.tostring). If you try to echo it, it'll be treated as a string. It is an object, however, so you can still access its properties.

In your view you should be able to access the enum instances' properties e.g.

```blade
<p>{{ $section_color->key }}<p>
<p>{{ $section_color->value }}<p>
<p>{{ $section_color->label }}<p>
```

**Comment by torian257x** [comment]
I just tried it in a brand new laravel project, and it does work as expected. That means the enum is not a string, everything just works fine.

I created an enum with integer backed values, and with string backed values, and both got added to the view as a fully fledged enum.

Soooo.... must be statamic then I guess

I just looked.

It's an issue from a statamic addon

https://statamic.com/addons/silentz/blade


`\Edalzell\Blade\Augmentation\AugmentationViewServiceProvider` json serializes everything that can be json serialized (that is a bit too much imho)

I will close the issue here, and talk to the maker of that addon 

thank you
